cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/CruiseControlConfiguration.java

```diff
@@ -118,10 +123,42 @@ public CruiseControlConfiguration(Reconciliation reconciliation, Iterable<Map.En
     }
 
     /**
-     * @return Map Cruise Control's default configuration properties
+     * Filters out Cruise Control resource goals if the necessary resource capacityConfiguration settings are not defined in
+     * the user's Kafka custom resource.
+     *
+     * @param goalList The initial list of supported Cruise Control goals.
+     * @param capacityConfiguration The object containing Cruise Control capacity configuration information.
+     * @return A new list containing only the goals that are safe to enable given the configured resource capacities.
      */
-    public static Map<String, String> getCruiseControlDefaultPropertiesMap() {
-        return CRUISE_CONTROL_DEFAULT_PROPERTIES_MAP;
+    /* test */ static List<String> filterResourceGoalsWithoutCapacityConfig(List<String> goalList,
+                                                                            CapacityConfiguration capacityConfiguration) {
+        List<String> filteredGoalList = new ArrayList<>(goalList);
+        if (!capacityConfiguration.isInboundNetworkConfigured()) {
+            filteredGoalList.remove(CruiseControlGoals.NETWORK_INBOUND_USAGE_DISTRIBUTION_GOAL.toString());
+            filteredGoalList.remove(CruiseControlGoals.NETWORK_INBOUND_CAPACITY_GOAL.toString());
+            filteredGoalList.remove(CruiseControlGoals.LEADER_BYTES_IN_DISTRIBUTION_GOAL.toString());
+        } else {
+            if (!capacityConfiguration.isCapacityHomogeneouslyConfigured(InboundNetworkCapacity.KEY)) {
+                /*
+                 * The LeaderBytesInDistributionGoal does not account for heterogeneous inbound network capacities.
+                 * See: https://github.com/linkedin/cruise-control/blob/main/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/LeaderBytesInDistributionGoal.java
+                 * Therefore, if the INBOUND_NETWORK capacityConfiguration settings are not homogeneously configured,
+                 * we remove this goal from the list.
+                 */
+                filteredGoalList.remove(CruiseControlGoals.LEADER_BYTES_IN_DISTRIBUTION_GOAL.toString());
+            }
+        }
+        if (!capacityConfiguration.isOutboundNetworkConfigured()) {
+            filteredGoalList.remove(CruiseControlGoals.NETWORK_OUTBOUND_USAGE_DISTRIBUTION_GOAL.toString());
+            filteredGoalList.remove(CruiseControlGoals.NETWORK_OUTBOUND_CAPACITY_GOAL.toString());
+            filteredGoalList.remove(CruiseControlGoals.POTENTIAL_NETWORK_OUTAGE_GOAL.toString());
+        }
+        if (!capacityConfiguration.isCpuConfigured()) {
+            filteredGoalList.remove(CruiseControlGoals.CPU_CAPACITY_GOAL.toString());
+            filteredGoalList.remove(CruiseControlGoals.CPU_USAGE_DISTRIBUTION_GOAL.toString());
+        }
```

Comment:
Slightly related to the other comment thread -> why do we need to remove the distribution tasks? How do they depend on the capacity?

Reply:
Ok, makes sense I guess. Thanks.